aws/resource_aws_servicecatalog_product_test.go

```diff
@@ -0,0 +1,241 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/servicecatalog"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccAWSServiceCatalogProduct_basic(t *testing.T) {
+	arbitraryProductName := fmt.Sprintf("product-%s", acctest.RandString(5))
+	arbitraryProvisionArtifactName := fmt.Sprintf("pa-%s", acctest.RandString(5))
+	arbitraryBucketName := fmt.Sprintf("bucket-%s", acctest.RandString(16))
+	tag1 := "FooKey = \"bar\""
+	tag2 := "BarKey = \"foo\""
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckServiceCatalogProductDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAwsServiceCatalogProductResourceConfigTemplate(arbitraryBucketName, arbitraryProductName, arbitraryProvisionArtifactName, tag1, tag2),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("aws_servicecatalog_product.test", "name", arbitraryProductName),
+					resource.TestCheckResourceAttr("aws_servicecatalog_product.test", "provisioning_artifact.#", "1"),
+					resource.TestCheckResourceAttr("aws_servicecatalog_product.test", "provisioning_artifact.0.name", arbitraryProvisionArtifactName),
+					resource.TestCheckResourceAttrSet("aws_servicecatalog_product.test", "provisioning_artifact.0.description"),
+					resource.TestCheckResourceAttr("aws_servicecatalog_product.test", "tags.%", "2"),
+					resource.TestCheckResourceAttr("aws_servicecatalog_product.test", "tags.FooKey", "bar"),
+					resource.TestCheckResourceAttr("aws_servicecatalog_product.test", "tags.BarKey", "foo"),
+					resource.TestCheckResourceAttrSet("aws_servicecatalog_product.test", "description"),
+					resource.TestCheckResourceAttrSet("aws_servicecatalog_product.test", "distributor"),
+					resource.TestCheckResourceAttrSet("aws_servicecatalog_product.test", "owner"),
+					resource.TestCheckResourceAttrSet("aws_servicecatalog_product.test", "product_type"),
+					resource.TestCheckResourceAttrSet("aws_servicecatalog_product.test", "support_description"),
+					resource.TestCheckResourceAttrSet("aws_servicecatalog_product.test", "support_email"),
+					resource.TestCheckResourceAttrSet("aws_servicecatalog_product.test", "support_url"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSServiceCatalogProduct_updateTags(t *testing.T) {
```

Comment:
can you rewrite this based on https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/contribution-checklists.md#resource-tagging-acceptance-testing-implementation